src/settings/SettingsStore.js

```diff
@@ -138,21 +139,15 @@ export default class SettingsStore {
         }
 
         const watcherId = `${new Date().getTime()}_${settingName}_${roomId}`;
-        SettingsStore._watchers[watcherId] = {};
-
-        const levels = Object.keys(LEVEL_HANDLERS);
-        for (const level of levels) {
-            const handler = SettingsStore._getHandler(originalSettingName, level);
-            if (!handler) continue;
 
-            const localizedCallback = (changedInRoomId, newVal) => {
-                callbackFn(originalSettingName, changedInRoomId, level, newVal);
-            };
+        const localizedCallback = (changedInRoomId, atLevel, newValAtLevel) => {
+            const newValue = SettingsStore.getValue(originalSettingName);
+            callbackFn(originalSettingName, changedInRoomId, atLevel, newValAtLevel, newValue);
+        };
 
-            console.log(`Starting watcher for ${settingName}@${roomId || '<null room>'} at level ${level}`);
-            SettingsStore._watchers[watcherId][level] = localizedCallback;
-            handler.watchSetting(settingName, roomId, localizedCallback);
-        }
+        console.log(`Starting watcher for ${settingName}@${roomId || '<null room>'}`);
+        SettingsStore._watchers[watcherId] = localizedCallback;
+        defaultWatchManager.watchSetting(settingName, roomId, localizedCallback);
```

Comment:
You could wonder now if the WatchManager is still needed. It only ever has one watcher for a setting, so it might further simplify by bringing it into SettingsStore and merging it with _watchers. Handlers could just call `notifyUpdate` on SettingsStore.

Reply:
I kinda want to keep as much as possible out of the store so it doesn't become a 1000 line monster. Ideally I'd also move a bunch of the watcher stuff out, but this is fine for now I think.